lib/train-habitat/connection.rb

```diff
@@ -0,0 +1,44 @@
+require 'net/http'
+require 'json'
+require 'train-habitat/platform'
+
+module TrainPlugins
+  module Habitat
+    class Connection < Train::Plugins::Transport::BaseConnection
+      include TrainPlugins::Habitat::Platform
+
+      def initialize(options)
+        msg = 'Habitat host serving HTTP Gateway is required'
+        raise Train::TransportError, msg if options.nil? || options[:host].nil?
+
+        super(options)
+      end
+
+      def uri
+        "habitat://#{@options[:host]}"
+      end
+
+      # Example usage in an InSpec Resource Pack: inspec.backend.habitat_client.services
+      def habitat_client
+        klass = HTTPGateway
+
+        return klass.new(@options[:host]) unless cache_enabled?(:api_call)
+        @cache[:api_call][klass.to_s.to_sym] ||= klass.new(@options[:host])
+      end
```

Comment:
The local variable just points to a constant, and ...doesn't vary. So why use it?

Also, `dig` is preferred for descending past one level in a hash unless you're absolutely certain all of the keys will be there.
```ruby
def habitat_client
  return HTTPGateway.new(@options[:host]) unless cache_enabled?(:api_call)

  @cache.dig(:api_call, :HTTPGateway) ||= HTTPGateway.new(@options[:host])
end
```